tools/shoestring/shoestring/healthagents/rest_https_certificate.py

```diff
@@ -27,39 +27,63 @@ def _openssl_run_sclient_verify(hostname, test_args):
 	] + test_args, command_input='Q', show_output=False)
 
 	# verify the certificate is valid
+	certificates = []
+	collect_certificates = False
 	for line in lines:
 		line = line.strip()
 		if line.startswith('verify error:'):
 			return False, line
 
+		# collect all certificates
+		if line.startswith('-----BEGIN CERTIFICATE-----'):
+			collect_certificates = True
+
+		if collect_certificates:
+			certificates.append(line + '\n')
+
+		if line.startswith('-----END CERTIFICATE-----'):
+			collect_certificates = False
+
 	with tempfile.TemporaryDirectory() as temp_directory:
 		temp_file = Path(temp_directory) / 'cert.txt'
 		with open(temp_file, 'wt', encoding='utf-8') as cert_outfile:
-			cert_outfile.writelines(lines)
-
+			cert_outfile.writelines(certificates)
 		lines = openssl_executor.dispatch([
-			'x509',
-			'-inform', 'pem',
-			'-noout',
+			'storeutl',
 			'-text',
-			'-in', temp_file
+			'-noout',
+			'-certs',
+			temp_file
 		], command_input='Q', show_output=False)
 
-	date_patterns = [r'Not Before: (.*)', r'Not After : (.*)']
+	date_patterns = []
+	cert_dates = []
 	collect_dates = False
 	collected_dates = []
 	for line in lines:
 		line = line.strip()
 		if collect_dates and date_patterns:
 			res = re.search(date_patterns[0], line)
 			if res:
-				collected_dates.append(parsedate_to_datetime(res.group(1)))
+				cert_dates.append(parsedate_to_datetime(res.group(1)))
 				date_patterns.pop(0)
+				if not date_patterns:
+					# we have collected both dates for a certificate
+					collected_dates.append((cert_dates[0], cert_dates[1]))
+					cert_dates.clear()
+					collect_dates = False
 
-		if line == 'Certificate:':
+				continue
+
+		if 'Certificate:' == line:
 			collect_dates = True
+			date_patterns = [r'Not Before: (.*)', r'Not After : (.*)']
+
+		if line.startswith('Total found:'):
+			if f'Total found: {len(collected_dates)}' == line:
+				return True, collected_dates
```

Comment:
why do you need this condition vs just returning true at end if len(collected_dates) != 1

Reply:
Currently, it verifies that the certificates found by OpenSSL match the number of dates collected.
If this is a best-effort scenario, then it can check that at least one date is collected.